### DISCUSSION TITLE
How to extend DocumentSettings Components

### DISCUSSION THREAD

**Discussion by ibes** [WordPress/gutenberg#40283] [Q&A]
To make it a clear example:

I would like to add a character count to the excerpt field.

In Classic Editor I would do a litte jQuery for that. 
But in Gutenberg I need to get inside the React Component, right?

Is there a way to extend that components?
I heard of Higher Level Components and imagine to wrap the original compoent to extend it with further features?!

I am React beginner and hope to learn my way to have full control over the admin panel again.

**Comment by t-hamano** [comment]
Hi @ibes,

Gutenberg has pre-defineds place for developers to inject items, which can be achieved by using SlotFills.
[SlotFills Reference | Block Editor Handbook | WordPress Developer Resources](https://developer.wordpress.org/block-editor/reference-guides/slotfills/)

For example, `PluginPostStatusInfo` may probably suit your purposes.
[PluginPostStatusInfo | Block Editor Handbook | WordPress Developer Resources](https://developer.wordpress.org/block-editor/reference-guides/slotfills/plugin-post-status-info/)

The `getEditedPostAttribute` may be used to retrieve the excerpt string to count the number of characters.
[gutenberg/data-core-editor.md at trunk · WordPress/gutenberg · GitHub](https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/data/data-core-editor.md#geteditedpostattribute)

The following references may be helpful for basic instructions on how to create a plugin to extend the block editor.
[How to use JavaScript with the Block Editor | Block Editor Handbook | WordPress Developer Resources](https://developer.wordpress.org/block-editor/how-to-guides/javascript/)
